packages/annotations/src/ui-components/comment.vue

```diff
@@ -0,0 +1,86 @@
+<template>
+  <div class="alpheios-annotations-comment__container" data-annotation-container>
+    <div class="alpheios-annotations__act-panel">
+      <div class="alpheios-annotations__act-ctrls" :data-annotation-form-open="isFormOpen">
+        <div class="alpheios-annotations__act-ctrls-edit" @click.stop="editComment"
+             :data-annotation-selected="isEditing">[*comment]</div>
+        <div class="alpheios-annotations__act-ctrls-remove" @click.stop="removeComment"
+             :data-annotation-selected="isRemoving">[-remove]</div>
+      </div>
+
+      <div class="alpheios-annotations__act-form" v-show="isEditing">
+        <div class="alpheios-annotations__act-form-content">
+          <div class="alpheios-annotations__act-form-headline">Edit the comment:</div>
+          <div class="alpheios-annotations__act-form-text-input-group">
+            <textarea id="lemma-definition" cols="50"
+                      name="lemma-definition" rows="4">A comment about the part of speech of the lexeme</textarea>
+          </div>
+        </div>
+
+        <div class="alpheios-annotations__act-form-ctrls">
+          <div class="alpheios-annotations__act-form-ctrls-save" @click.stop="closeForm">Save</div>
+          <div class="alpheios-annotations__act-form-ctrls-cancel" @click.stop="closeForm">Cancel</div>
+        </div>
+      </div>
+
+      <div class="alpheios-annotations__act-form" v-show="isRemoving">
+        <div class="alpheios-annotations__act-form-content">
+          <div class="alpheios-annotations__act-form-headline">Remove the comment:</div>
+          <div class="alpheios-annotations__act-form-text">
+            Are you sure that you want to remove this comment?
+          </div>
+        </div>
+        <div class="alpheios-annotations__act-form-ctrls">
+          <div class="alpheios-annotations__act-form-ctrls-remove" @click.stop="closeForm">Remove</div>
+          <div class="alpheios-annotations__act-form-ctrls-cancel" @click.stop="closeForm">Cancel</div>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+<script>
+const ActionTypes = {
+  NONE: 'none',
+  EDIT: 'edit',
+  REMOVE: 'remove'
+}
+
+export default {
+  name: 'AnnotationComment',
+  data () {
+    return {
+      selectedAction: ActionTypes.NONE
+    }
+  },
+  computed: {
```

Comment:
Maybe we should have a base component that has all of the basic methods (isAdding, isRemoving, isFormOpen) that the individual components can be derived from? This way we could make sure the functionality that is common is stable across all and avoid code duplication?

Reply:
I was thinking about that too, but I decided to postpone this for later. Right now components are simple because they do almost nothing. I was afraid that once we start to put more complex logic into them, they would start to drift away from each other. But I agree the common base class is a great idea. Will add that.

Reply:
ok. I'm fine with it being postponed to another PR